Add buttons: removing styles for the tiny plus sign

rickybanister opened the pull request:
I removed the proprietary styles for the 'add something' icon combinations.

I then added a global rule to nudge any plus-sign icons closer to an adjacent gridicon like 'add a person' or 'add a plugin'.

Also cleaned up some unused and/or weird styles relating to those existing button instances.

**Before:**
<img width="110" alt="screen shot 2016-03-15 at 12 00 18 pm" src="https://cloud.githubusercontent.com/assets/437258/13784427/940e1c4c-eaa5-11e5-9616-ad2c44885677.png">
<img width="203" alt="screen shot 2016-03-15 at 12 00 36 pm" src="https://cloud.githubusercontent.com/assets/437258/13784428/9420ef2a-eaa5-11e5-9b5d-33675e60025e.png">


**After:**
<img width="486" alt="screen shot 2016-03-15 at 11 34 24 am" src="https://cloud.githubusercontent.com/assets/437258/13784377/66cf3f2c-eaa5-11e5-9a7e-965b080e01c3.png">
<img width="202" alt="screen shot 2016-03-15 at 11 34 41 am" src="https://cloud.githubusercontent.com/assets/437258/13784379/66d1ee3e-eaa5-11e5-846f-b21e974e14f8.png">
<img width="769" alt="screen shot 2016-03-15 at 11 59 26 am" src="https://cloud.githubusercontent.com/assets/437258/13784378/66cff39a-eaa5-11e5-9036-6d7035a16c92.png">

cc @mtias @MichaelArestad

mtias commented:
I like it. Simple and scales. :ship:

MichaelArestad commented:
The code looks good, but the huge amount of white space on the left side of the button is a bit funky.

rickybanister commented:
I hinted the left margin over and moved the styles to buttons' css, and rebased.

<img width="91" alt="screen shot 2016-03-15 at 4 14 45 pm" src="https://cloud.githubusercontent.com/assets/437258/13792840/45875014-eaca-11e5-9ae8-5c93b15cd928.png">

The whitespace to the left of the button looks like the above now.

ebinnion commented on client/components/button/style.scss:
```diff
@@ -76,6 +76,14 @@ button {
 			margin-top: -8px;
 		}
+		// Make the left margin of the small plus icon visually less huge
+		.gridicons-plus-small {
```
Will there ever be a condition in which we'd only use the `+` icon? If so, would this make that icon be off-center?

rickybanister replied:
I could add something like this next to that

`.gridicon-plus-small:last-child { margin-left: 0; }`

To set it back to 0 if there was not a gridicon next to it, but that's obviously a little ugly. Thoughts @mtias or @MichaelArestad ?

mtias replied:
Perhaps with `:only-child`?

rickybanister replied:
Right, of course.

rickybanister replied:
`:only-child` won't work because there's a `noscript` tag inside that button. What about using `:last-of-type`?

ebinnion commented:
Other than the potential issue I commented about above, this seems to work well :+1:

rickybanister commented:
I added a `:last-of-type` rule to reset the margin on the plus icon when it appears as the final icon in a compact button. So if used by itself it has equal margins, if used next to another icon it has nicely hinted margins.

mtias commented:
:ship:
